src/API/FactoryCollection.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\API;
+use ILIAS\API\Int as I;
+
+/**
+ * Collection of factories
+ *
+ * @author killing@leifos.de
+ */
+class FactoryCollection implements I\FactoryCollection
```

Comment:
I like the idea of this factory stack created by the construction order of the factories, to determine parent/child relations when dispatching the command later on.